builtin/logical/database/path_config_connection.go

```diff
@@ -214,19 +232,46 @@ func (b *databaseBackend) connectionDeleteHandler() framework.OperationFunc {
 // both builtin and plugin database types.
 func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-		pluginName := data.Get("plugin_name").(string)
-		if pluginName == "" {
-			return logical.ErrorResponse(respErrEmptyPluginName), nil
-		}
+		verifyConnection := data.Get("verify_connection").(bool)
 
 		name := data.Get("name").(string)
 		if name == "" {
 			return logical.ErrorResponse(respErrEmptyName), nil
 		}
 
-		verifyConnection := data.Get("verify_connection").(bool)
-		allowedRoles := data.Get("allowed_roles").([]string)
-		rootRotationStatements := data.Get("root_rotation_statements").([]string)
+		// Baseline
+		config := &DatabaseConfig{}
+
+		entry, err := req.Storage.Get(ctx, fmt.Sprintf("config/%s", name))
+		if err != nil {
+			return nil, errors.New("failed to read connection configuration")
+		}
+		if entry != nil {
+			if err := entry.DecodeJSON(config); err != nil {
+				return nil, err
+			}
+		}
+
+		if pluginNameRaw, ok := data.GetOk("plugin_name"); ok {
+			config.PluginName = pluginNameRaw.(string)
+		} else if req.Operation == logical.CreateOperation {
+			config.PluginName = data.Get("plugin_name").(string)
```

Comment:
In all of these `else if` blocks, I think you can just initialize to the appropriate zero value ("" in this case). We already know the parameter doesn't exist and don't need to check it again.

Reply:
The reason we usually use this is in case we decide to provide/change a default value later. Instead of setitng the default in the schema *and* here, we just set it in the schema.

Reply:
Ah, TIL... That makes sense.